### PR TITLE
[TGBOT-279] Add Jacoco, SonarCloud and checkstyle

### PR DESCRIPTION
**Pull request by schneidervs**

### CODE REVIEW

**Comment by schneidervs** [comment]
Jacoco reports are not in the Sonar

**Review by yurasava — APPROVED** [review]

File: integration/pom.xml [hunk 1]
```diff
@@ -39,7 +39,7 @@
                         <goals>
                             <goal>report-aggregate</goal>
                         </goals>
-                        <phase>verify</phase>
+                        <phase>package</phase>
```
Comment (max-java) [inline integration/pom.xml:42]: I would not agree on this.
Tools like Sonar and Checkstyle are part of verifying phase, not package. 
we do packaging when everything is ok. 

What is the reason behind this changes?

**Review by max-java — CHANGES_REQUESTED** [review]
ptal

**Comment by max-java** [comment]
any updates on this?

File: integration/pom.xml (hunk 1 above)
Comment (schneidervs) [reply integration/pom.xml:42]: Sonar won't get a coverage report until we correct all the deficiencies. To get a coverage report, but drop the project if there are errors.

**Review by AlexanderYarokhovich — APPROVED** [review]

**Review by max-java — APPROVED** [review]

**Review by Sav4ykVladzislau — APPROVED** [review]

**Review by Artem-Metlushko — APPROVED** [review]

**Pull request merged**